sql/core/src/main/scala/org/apache/spark/sql/Dataset.scala

```diff
@@ -390,7 +390,7 @@ class Dataset[T] private[sql](
    */
   // This is declared with parentheses to prevent the Scala compiler from treating
   // `ds.toDF("1")` as invoking this toDF and then apply on the returned DataFrame.
-  def toDF(): DataFrame = new Dataset[Row](sparkSession, queryExecution, RowEncoder(schema))
+  def toDF(): DataFrame = new Dataset[Row](sparkSession, planWithBarrier, RowEncoder(schema))
```

Comment:
This will probably create a huge regression if you rerun the `DataFrame`. SparkPlans cache shuffle RDDs and we reuse previously executed shuffles when you re-run the frame.